swap/src/network/spot_price.rs

```diff
@@ -40,20 +40,34 @@ pub struct Request {
 }
 
 #[derive(Serialize, Deserialize, Debug, Clone)]
-pub struct Response {
-    pub xmr: monero::Amount,
+pub enum Response {
+    Xmr(monero::Amount),
+    Error(Error),
 }
```

Comment:
Can we have a form of snapshot test for this to make sure we don't accidentially break the wire-format? (i.e. serialize this and test against a hardcoded serialization).

Depending on your familiarity with serde, I would recommend that in general so you have a mental model of what the data you are sending actually looks like.

Reply:
I would have to write these tests against a `cbor` snapshot given that we use `cbor` encoding, correct?